test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -329,70 +395,90 @@ public void testPyScikitKMeans() throws Exception {
         String clustObj1Name = "py_clusters1";
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
                 .withData(getSampleMatrix());
-        WorkspaceClient wscl = getWsClient();
-        wscl.saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        List<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String, String>>> getInfoEKN = getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(exprObjName).withType("KBaseFeatureValues.ExpressionMatrix")
-                .withData(new UObject(data)))));
-        impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" + 
+                        .withData(new UObject(data)))));
+        String getId = impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" +
                 exprObjName).withK(3L).withOutWorkspace(testWsName).withOutClustersetId(clustObj1Name),
                 token, getContext());
-        ObjectData res = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+        ObjectData res = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
                 .withName(clustObj1Name))).get(0);
+
+        //test if returned id matches id returned by ws saveobjects
+        Assert.assertEquals(getInfoEKN.get(0).getE7()  + "/" + (getInfoEKN.get(0).getE1()+1) + "/" +  getInfoEKN.get(0).getE5(), getId);
```

Comment:
It's somewhat confusing that in some cases the tests check the ref from an object info tuple vs the ref returned by the method, and some tests increment either the object ID or version by 1. If possible I'd choose one or the other.